### PR TITLE
Clean up compiler of MSVC 2013 conditional code

### PR DESCRIPTION
**Pull request by bhardwajs**
Clean up MSVC 2013 conditional code in C++ codegen. Also change the gbc
tests as per the codegen. To generate the tests:
> cd compiler
>  stack test :gbc-tests --test-arguments --accept

Note that this PR can only be merged after #1030 is merged.

### CODE REVIEW

**Comment by chwarr** [comment]
I merged your PR that removed MSVC 2013 from the build matrix. Can you merge with the current tip of master so that we can get a clean build?

File: compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs [hunk 1]
```diff
@@ -308,11 +305,7 @@ types_h export_attribute userHeaders enumHeader allocator alloc_ctors_enabled ty
             -- even if implicit would be okay, fall back to explicit for
             -- compilers that don't support = default for move constructors
```
Comment (chwarr) [inline compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs:306]: Minor: comment looks stale

File: compiler/src/Language/Bond/Codegen/Cpp/Enum_h.hs [hunk 2]
```diff
@@ -20,7 +20,7 @@ enum_h :: MappingContext -> String -> [Import] -> [Declaration] -> (String, Text
 enum_h cpp _file _imports declarations = ("_enum.h", [lt|
 #pragma once
 
-#include <stdint.h>
+#include <cstdint>
```
Comment (chwarr) [inline compiler/src/Language/Bond/Codegen/Cpp/Enum_h.hs:23]: While `<cstdint>` may be more C++ish, we use the unnamespaced `<stdint.h>` type names everywhere else and consuming code may be doing so as well. This doesn't seem like we get enough benefit to make that breaking change.

File: compiler/src/Language/Bond/Codegen/Cpp/Util.hs [hunk 3]
```diff
@@ -159,8 +159,8 @@ enumDefinition Enum {..} = [lt|enum #{declName}
         };|]
   where
     constant Constant {..} = [lt|#{constantName}#{optional value constantValue}|]
-    value (-2147483648) = [lt| = static_cast<int32_t>(-2147483647-1)|]
-    value x = [lt| = static_cast<int32_t>(#{x})|]
+    value (-2147483648) = [lt| = static_cast<std::int32_t>(-2147483647-1)|]
```
Comment (chwarr) [inline compiler/src/Language/Bond/Codegen/Cpp/Util.hs:162]: See other comment: I think we want to revert this change.

File: compiler/src/Language/Bond/Codegen/Cpp/Enum_h.hs (hunk 2 above)
Comment (bhardwajs) [reply compiler/src/Language/Bond/Codegen/Cpp/Enum_h.hs:23]: I looked through the current versions of `<cstdint>` in MSVC, gcc, and clang and all three would be fine with this code.

However, you are write, the consuming code might not be on the most recent toolchain. I will revert.
